.github/workflows/main.yml

```diff
@@ -4,41 +4,31 @@ on:
   push:
     branches: [main]
 
-env:
-  package: "@techdocs/cli"
-
 jobs:
-  precheck:
-    runs-on: ubuntu-latest
-    outputs:
-      isPublished: ${{ steps.npm.outputs.view }}
-    steps:
-    - uses: actions/checkout@v2
-    - id: npm
-      run: |
-        CURRENT_VERSION=$(cat packages/techdocs-cli/package.json \
-          | grep version \
-          | head -1 \
-          | awk -F: '{ print $2 }' \
-          | sed 's/[",]//g' \
-          | tr -d '[[:space:]]')
-        NPM_VIEW=$(npm view ${{ env.package }}@$CURRENT_VERSION --json)
-        echo "::set-output name=view::$NPM_VIEW"
-
   release:
+    name: Release
     runs-on: ubuntu-latest
-    needs: precheck
-    if: ${{ !needs.precheck.outputs.isPublished }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - name: Checkout Repo
+        uses: actions/checkout@master
+        with:
+          # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+
+      - name: Setup Node.js 12.x
+        uses: actions/setup-node@master
         with:
           node-version: 12.x
           registry-url: https://registry.npmjs.org/ # Needed for auth
-      - run: yarn install --frozen-lockfile
-      - run: yarn tsc
-      - run: yarn build
-      - name: Publish
-        run: yarn workspace ${{ env.package }} publish --access public
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@master
```

Comment:
how does this action know if it should open a new PR or publish? (just thinking as the precheck is deleted)

Reply:
It pulls the current version of npm

Reply:
https://github.com/atlassian/changesets/blob/main/packages/cli/src/commands/publish/publishPackages.ts#L203

Reply:
This is what the action does in general terms:

1. If you merge a pull request containing changeset files, the action will create a 'Version Package' PR if it does not exist, or append the merged changes to 'Version Pack' PR if it already exists.

2. The `Package Version` pull request deletes all changeset files, so when we merge it, the action will see that we have no changeset files, pulls the package versions from npm, checks for the local version hasn't been published yet, and if so, the action publish a new version, generate and submit a tag, and release in the package repository.